app/Http/Requests/PostRequest.php

```diff
@@ -26,12 +26,12 @@ public function rules()
         }
 
         return [
-            'campaign_id' => 'required|integer',
+            'campaign_id' => 'required_without:action_id|integer',
             'northstar_id' => 'nullable|objectid',
-            'type' => 'required|string|in:photo,voter-reg,text,share-social',
+            'type' => 'required_without:action_id|string|in:photo,voter-reg,text,share-social',
```

Comment:
hi @DFurnes ! Continuing on from [your comment](https://github.com/DoSomething/rogue/pull/841#discussion_r261268220), right now, I don't think we actually require a photo post to have a file (somewhere this got lost unless I wasn't looking hard enough for this code!). This might make it a little more difficult to have the validation in the post request but I was thinking we could throw an error later, after we make the db query anyway for the type of post. But that probably isn't ideal... 

Re: "
and we [allow different statuses](https://github.com/DoSomething/rogue/blob/4a12be9f680f7cec776be1e81573357440a0cd5a/app/Http/Requests/PostRequest.php#L66-L77) based on the type of post as well", this is just for `PATCH` requests, in which case we already have the post type from the post object so I don't think this would be an issue here!

Reply:
> Continuing on from [your comment](https://github.com/DoSomething/rogue/pull/841#discussion_r261268220), right now, I don't think we actually require a photo post to have a file (somewhere this got lost unless I wasn't looking hard enough for this code!).

Yup, you're right. We definitely should, though.

> this is just for `PATCH` requests, in which case we already have the post type from the post object

Hmm, are you sure? It looks like we use this for [post creation as well](https://github.com/DoSomething/rogue/blob/4a12be9f680f7cec776be1e81573357440a0cd5a/app/Http/Requests/PostRequest.php#L40) to me.

Reply:
Oh shoot you're totally right, @DFurnes ! I missed that line! Ok so I think it is better to revert back to still requiring `type` then, cc @rapala61 @aaronschachter . 

@DFurnes to revert this PR back, can I just git rebase and drop all the new commits that adds this work? 

Also, [here](https://www.pivotaltracker.com/n/projects/2019429/stories/164308611) is a card to make sure we are actually requiring the `file` when a `photo` post is created!

Reply:
@chloealee reading through the comments. Am I understanding correctly that type will continue to be expected regardless of sending `action_id`? Also, another question related to the changes I have to make in [Gateway-js](https://github.com/DoSomething/gateway-js/blob/master/server/src/endpoints/rogue/posts.js#L62). I'll move that line to Gambit so that in the event `action_id` is not included in the payload, `action: 'default'` will be sent instead. Is this inn line with the changes in this PR?

Reply:
> Am I understanding correctly that `type` will continue to be expected regardless of sending `action_id`

@rapala61 yes that is correct! We do some validation around the post type (so if it is a photo post, we want to validate that there is a `file` in the payload as well) which we want to have this type handy instead of doing a DB call. 

> I'll move that line to Gambit so that in the event `action_id: 1234` is not included in the payload, `action: 'default'` will be sent instead. Is this in line with the changes in this PR?

Yes! This is a little tricky though because if the action has a different name than `default`, it won't find the action in the query. I think the work around right now is that we're creating separate actions for SMS posts as with the name of `default`. I think this is how we have to go for now if there is no `action_id` because there is no way for a user to enter this info in gambit I think? cc @ngjo

Reply:
Yup - I'm working w/ Anthony tomorrow to make sure he's setting up action ids in Rogue for SMS for the time being to have `default` so things work!

Reply:
Thanks for confirming, @ngjo !

Reply:
So to wrap up this conversation, we need to revert the `required_without:action_id` on this line to `required`!

Reply:
ah yes, thanks for catching that, @DFurnes ! I was rebasing and it must have changed back then! Updated :)